gm4_metallurgy/data/gm4_vibro_shamir/functions/shock/user.mcfunction

```diff
@@ -0,0 +1,22 @@
+# @s = player who used vibro
+# run from shock/activate
+
+# set up player
+tag @s add gm4_vibro_user
+scoreboard players operation $damage gm4_vibro_shock = @s gm4_vibro_shock
+
+# calculate damage dealt to mobs
+execute as @e[team=!invalid_team,type=!armor_stand,tag=!gm4_vibro_user,distance=..6] at @s run function gm4_vibro_shamir:shock/calculate
```

Comment:
as written above, the `team=!invalid_team` is used to target living mobs. If we don't want flying mobs, then it'd be better to test for `nbt={OnGround:1b}`, so mobs not on the ground won't be affected. This would also make it so players can avoid the attack in a PVP setting